### PR TITLE
Feature/cleanup

### PR DESCRIPTION
**Pull request by aerorahul**
This PR:
- adds a .gitignore file
- adds a README.md file
- adds a VERSION file that provides version number of the library
- cleans up CMakeLists.txt

The CMakeLists.txt clean-up includes:
- adds a version to the project, along with languages
- uses standard build types and checks for supported compilers.
- removal of globbing of source files
- using a CMake defined variables where appropriate
- splits compiler flags into common, debug, release
- uses `add_compile_definitions` where appropriate instead of `compile_flags`
- removes `POSITION_INDEPENDENT_CODE`.  Libraries are built as `STATIC`. If `SHARED` is desired, this feature is automatically added by cmake.
- formatting using `cmake-format`

### CODE REVIEW

File: VERSION
Comment (fossell) [inline VERSION]: What does this version refer to?

**Review by fossell — CHANGES_REQUESTED** [review]
Changes in this PR do not impact source code. Ok with me but wondering what "VERSION" (v1.1.1) is in reference to?

File: VERSION
Comment (aerorahul) [reply VERSION]: @fossell In the `CMakeLists.txt`, a version number `1.1.1` was hard-wired.  This PR removes the hard-wiring from the `CMakeLists.txt`, and places it in this file.  If this is an inappropriate number, please suggest what is the appropriate version that this application is at.

File: CMakeLists.txt
Comment (aerorahul) [inline CMakeLists.txt]: use `VERSION` file in obtaining the version number of the application.

File: VERSION
Comment (fossell) [reply VERSION]: Thanks for clarifying, post is using v8.0.0 for this app.

Comment (aerorahul) [reply VERSION]: @fossell version number has been update to 8.0.0

**Review by fossell — APPROVED** [review]

**Comment by fossell** [comment]
@aerorahul - I can merge this PR once at least one UFS team member (@climbfuji @kgerheiser @mark-a-potts) reviews and approves these mods.

**Comment by climbfuji** [comment]
This PR needs to be tested together with all other cleanup PRs on a large number of machines using the NCEPLIBS umbrella build and CIME workflow before it can be merged. In particular: hera, gaea, generic macOS, generic linux, possibly also orion and stampede.

**Comment by fossell** [comment]
@climbfuji - Understood, will wait until testing is complete and approval is given.

**Comment by aerorahul** [comment]
@wenmeng-NOAA please revert the merge. This PR is a part of larger set of PRs. They all go in or none of them go in kind of a deal. Thanks!

**Comment by WenMeng-NOAA** [comment]
@aerorahul It seems you have to revert from your folk, see:
https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/reverting-a-pull-request

**Comment by WenMeng-NOAA** [comment]
@aerorahul A new PR was created for reverting. Please review it. Thanks!

**Comment by fossell** [comment]
@aerorahul - I know this was reverted, but does this clean up need to be merged still ?  Has the work been completed?

**Comment by aerorahul** [comment]
@fossell - I think it was re-merged. I don't see any differences between `release/public-v8` and `feature/cleanup` from my fork.
This work is now complete.

**Comment by fossell** [comment]
@aerorahul - Thanks for confirming!

**Pull request merged**
